### PR TITLE
Consumed airtime metadata in uplink messages

### PR DESCRIPTION
**Pull request by neoaggelos**
#### Summary

References #35 

#### Changes

- Add `consumed_airtime` field in `ApplicationUplink` messages
- NS calculates and adds consumed airtime as metadata before forwarding message to Application Server
- Update documentation

#### Testing

- Test locally, calculation is covered by existing unit tests in the `pkg/toa` package

##### Regressions

...

#### Notes for Reviewers

...

#### Checklist

- [X] Scope: The referenced issue is addressed, there are no unrelated changes.
- [X] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [X] Documentation: Relevant documentation is added or updated.
- [X] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [X] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: pkg/networkserver/grpc_gsns.go [hunk 1]
```diff
@@ -996,19 +997,24 @@ func (ns *NetworkServer) handleDataUplink(ctx context.Context, up *ttnpb.UplinkM
 		log.FromContext(ctx).WithError(err).Error("Failed to update downlink task queue after data uplink")
 	}
 	if !matched.IsRetransmission {
+		var airtimep *time.Duration
+		if airtime, err := toa.Compute(len(up.RawPayload), up.Settings); err == nil {
+			airtimep = &airtime
+		}
```
Comment (rvolosatovs) [inline pkg/networkserver/grpc_gsns.go]:
```suggestion
		var airtime *time.Duration
		t, err := toa.Compute(len(up.RawPayload), up.Settings)
		if err != nil {
			log.FromContext(ctx).WithError(err).Debug("Failed to compute time-on-air")
		} else {
			airtime = &t
		}
```

It `toa.Compute` expected to fail? If so, when?

File: pkg/networkserver/grpc_gsns.go [hunk 2]
```diff
@@ -996,19 +997,24 @@ func (ns *NetworkServer) handleDataUplink(ctx context.Context, up *ttnpb.UplinkM
...
-					ReceivedAt:   up.ReceivedAt,
+					Confirmed:       up.Payload.MType == ttnpb.MType_CONFIRMED_UP,
+					FCnt:            pld.FullFCnt,
+					FPort:           pld.FPort,
+					FRMPayload:      pld.FRMPayload,
+					RxMetadata:      up.RxMetadata,
+					SessionKeyID:    stored.Session.SessionKeyID,
+					Settings:        up.Settings,
+					ReceivedAt:      up.ReceivedAt,
+					ConsumedAirtime: airtimep,
```
Comment (rvolosatovs) [inline pkg/networkserver/grpc_gsns.go]:
```suggestion
					ConsumedAirtime: airtime,
```

**Review by rvolosatovs — CHANGES_REQUESTED** [review]
Should this also be done on join requests?

**Review by htdvisser — APPROVED** [review]
Nothing to add

File: pkg/networkserver/grpc_gsns.go (hunk 1 above)
Comment (johanstokking) [reply pkg/networkserver/grpc_gsns.go]: Invalid parameters. Probably not in this context.

**Review by johanstokking — APPROVED** [review]
Nothing to add either

**Review by rvolosatovs — APPROVED** [review]

**Pull request merged**